### PR TITLE
[interp] Split leave into pieces, partially outlined, to conserve stack,

### PR DESCRIPTION
**Pull request by jaykrell**
but without changing how exception handling macros access labels and locals.

This saves 16 bytes of stack on Linux/amd64/gcc and likely similar everywhere,
as it takes an address-taken local out of interp_exec_method_full (stackval tmp_sp).

See
https://github.com/mono/mono/pull/16171
https://github.com/mono/mono/pull/16171/commits/a6a8f0d5ffa875da9cda63444e401c37810a690f for a
full outlining.

Ultimately over 100 bytes are savable.
This is just one step toward that.

### CODE REVIEW

**Comment by jaykrell** [comment]
@monojenkins build failed

File: mono/mini/interp/interp.c [hunk 1]
```diff
@@ -2907,13 +2907,29 @@ static int opcode_counts[512];
 #define frame_objref(frame) o
 #endif
 
+// Leave is split into pieces in order to consume less stack,
+// but not have to change how exception handling macros access labels and locals.
+static MONO_NEVER_INLINE MonoException*
+mono_interp_leave (InterpFrame* child_frame)
```
Comment (BrzVlad) [inline mono/mini/interp/interp.c]: Maybe name this mono_interp_get_undeniable_exception or something like that. The current name is not really representative of what this method is doing.

Comment (jaykrell) [reply mono/mini/interp/interp.c]: To naïve/shallow/empirical me, it is `mono_interp_outlined_mechanical_part_of_leave_whatever_it_takes`, shortened to `mono_interp_leave`, not anything more general or more specific, not be used anywhere else. But of course I try to be flexible. :)

Comment (jaykrell) [reply mono/mini/interp/interp.c]: `mono_interp_leave_outlined`??
There might be another chunk, I was going to call them `mono_interp_leave1` and `mono_interp_leave2`.
I need to double check that it useful for purposes of stack.

Comment (jaykrell) [reply mono/mini/interp/interp.c]: I see what you mean though, it is one line doing just that.
Nitty question, then..
```					child_frame.imethod = NULL; 
					child_frame.imethod = NULL;
```
should be moved into this function, for it to be more complete/sensible per your name?
For stack preservation they can go in either place. I only moved the minimal amount, didn't attempt to make a sensible function.

File: mono/mini/interp/interp.c [hunk 2]
```diff
@@ -2907,13 +2907,29 @@ static int opcode_counts[512];
...
+{
+	stackval tmp_sp;
+	/*
+	 * We need for mono_thread_get_undeniable_exception to be able to unwind
+	 * to check the abort threshold. For this to work we use child_frame as a
+	 * dummy frame that is stored in the lmf and serves as the transition frame
+	 */
+	do_icall_wrapper (child_frame, NULL, MINT_ICALL_V_P, &tmp_sp, (gpointer)mono_thread_get_undeniable_exception, FALSE);
+
+	return (MonoException*)tmp_sp.data.p;
```
Comment (jaykrell) [inline mono/mini/interp/interp.c]: Feels like `p` should be `o` for wasm GC-safety. But I didn't change that.

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
Oops

**Comment by jaykrell** [comment]
@monojenkins build failed

**Pull request merged**